doc/manual/src/language/advanced-attributes.md

```diff
@@ -203,8 +203,8 @@ Derivations can declare some infrequently used optional attributes.
 
         This is the default.
 
-      - `"recursive"`\
-        The hash is computed over the NAR archive dump of the output
+      - `"nar"` or `"recursive"`\
```

Comment:
```suggestion
      - `"recursive"` or `"nar"`\
```

We may want to say that `recursive` is preferred because it's backward compatible. Indeed the downside of introducing such aliases is that they're not forward compatible.

Reply:
How is the note that I added?

I am not sure "recursive" is preferred. If anything I would hope it would be deprecated and people use "nar" for consistent and explicitness. But that is not a priority for me so I left not either being "preferred", just stating the facts so users can make their own choice.

Reply:
Using `nar` instead of `recursive` makes a Nix expression unnecessarily backwards-incompatible. E.g. if somebody adds a package that uses `nar` to Nixpkgs, it won't work on older versions of Nix while it otherwise probably would have. That's the danger of adding aliases like this.

Reply:
If we do it, I think it should warn loudly for at least 2 years and until all implementations support it, whichever is later.
If John wants to manage such a long term thing, I think we could let him. Before you know it, it'll be 2026 and time will have solved the problem.

Reply:
Yeah at the end of the day I am OK with warnings, docs saying don't use this, etc. The one thing I care about is that this also support `"nar"` so we could conceivably, many years from now, get the ecosystem off `"recursive"`.

Reply:
(I applied the suggestion and also reordered the explanations below.)